core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3412,7 +3413,7 @@ private boolean syncMetadata(RpcContext rpcContext, AlluxioURI path,
       boolean isGetFileInfo) throws AccessControlException, InvalidPathException {
     LockingScheme syncScheme = createSyncLockingScheme(path, options, isGetFileInfo);
 
-    if (!syncScheme.shouldSync()) {
+    if (mUfsAbsentPathCache.isAbsent(path) || !syncScheme.shouldSync()) {
```

Comment:
If a particular path got added to the absentPathCache and later that path is created out of band directly on the UFS, how do we discover that if we completely skip the sync?

Reply:
I think it will get processed and removed from the cache during the same ListStatus or GetStatus call but it happens too late in the process for the same call to return the right results.

Reply:
one idea @calvinjia has is to make it similar to the UfsStatusCache and respect sync interval?